packages/functional-tests/tests/syncV3/fxDesktopHandshake.spec.ts

```diff
@@ -84,80 +57,139 @@ test.describe('severity-2 #smoke', () => {
-      expect(await login.getEmailInput()).toContain('');
+      expect(await login.getPrefilledEmail()).toContain(email);
+      await login.setPassword(PASSWORD);
+      await login.clickSubmit();
+      expect(await login.isUserLoggedIn()).toBe(true);
     });
+  });
 
-    test('Sync force_auth page - user signed into browser is different to requested user', async ({
+  test.describe('Firefox desktop user info handshake', () => {
+    test.use({
+      emailOptions: [
+        { prefix: SIGNIN_EMAIL_PREFIX, password: PASSWORD },
+        { prefix: SYNC_EMAIL_PREFIX, password: PASSWORD },
+      ],
+    });
+
+    test('Non-Sync - user signed into browser, user signed in locally', async ({
       emails,
       target,
       syncBrowserPages: { login, page },
     }) => {
+      await Promise.all(
+        emails.map(async (email) => {
+          target.auth.signUp(email, PASSWORD, {
+            lang: 'en',
+            preVerified: 'true',
+          });
+        })
+      );
       const [email, syncEmail] = emails;
       const query = new URLSearchParams({
         forceUA: uaStrings['desktop_firefox_71'],
-        email: syncEmail,
       });
       const eventDetailStatus = createCustomEventDetail(
         FirefoxCommand.FxAStatus,
         {
-          signedInUser: email,
+          signedInUser: null,
         }
       );
       await page.goto(
-        `${
-          target.contentServerUrl
-        }?force_auth&automatedBrowser=true&context=fx_desktop_v3&service=sync&${query.toString()}`
+        `${target.contentServerUrl}?automatedBrowser=true&${query.toString()}`
       );
       await login.respondToWebChannelMessage(eventDetailStatus);
       await login.checkWebChannelMessage('fxaccounts:fxa_status');
+      await login.fillOutEmailFirstSignIn(syncEmail, PASSWORD);
+      expect(await login.isUserLoggedIn()).toBe(true);
+
+      // Then, sign in the user again, synthesizing the user having signed
+      // into Sync after the initial sign in.
+      const eventDetailStatusSignIn = createCustomEventDetail(
+        FirefoxCommand.FxAStatus,
+        {
+          signedInUser: email,
+        }
+      );
+      await page.goto(
+        `${target.contentServerUrl}?automatedBrowser=true&${query.toString()}`
+      );
+      await login.respondToWebChannelMessage(eventDetailStatusSignIn);
+      await login.checkWebChannelMessage('fxaccounts:fxa_status');
+
       expect(await login.getPrefilledEmail()).toContain(syncEmail);
+      await login.clickSignIn();
+      expect(await login.isUserLoggedIn()).toBe(true);
     });
 
-    test('Non-Sync force_auth page - user signed into browser is different to requested user', async ({
+    test('Sync force_auth page - user signed into browser is different to requested user', async ({
       emails,
       target,
       syncBrowserPages: { login, page },
     }) => {
+      await Promise.all(
+        emails.map(async (email) => {
```

Comment:
nit: might not need `async` since we aren't `await`ing or returning anything.
Which now has me 🤔 , what is this `await Promise.all()`ing if the `.map()` method isn't returning a promise?
Without a return/await, is this just quietly returning `[undefined, undefined]` which would `Promise.all()` resolve almost immediately (since the promises are running in parallel) and possibly have a race condition?

Reply:
Oops. I should be awaiting target.auth.signUp.